client/components/community-translator/index.jsx

```diff
@@ -0,0 +1,162 @@
+ */
+import Translatable from './translatable';
+import config from 'config';
+import User from 'lib/user';
+import userSettings from 'lib/user-settings';
+import { isCommunityTranslatorEnabled } from 'components/community-translator/utils';
+
+/**
+ * Local variables
+ */
+const debug = debugModule( 'calypso:community-translator' );
+const languages = config( 'languages' );
+const user = new User();
+
+class CommunityTranslator extends Component {
+	languageJson = null;
+	currentLocale = null;
+	initialized = false;
+
+	componentDidMount() {
+		this.setLanguage();
+
+		// wrap translations from i18n
+		i18n.registerTranslateHook( ( translation, options ) =>
+			this.wrapTranslation( options.original, translation, options )
+		);
+
+		// callback when translated component changes.
+		// the callback is overwritten by the translator on load/unload, so we're returning it within an anonymous function.
+		i18n.registerComponentUpdateHook( () => {} );
+		i18n.on( 'change', this.refresh );
+		user.on( 'change', this.refresh );
+		userSettings.on( 'change', this.refresh );
+	}
+
+	componentWillUnmount() {
+		i18n.off( 'change', this.refresh );
+		user.removeListener( 'change', this.refresh );
+		userSettings.removeListener( 'change', this.refresh );
+	}
+
+	setLanguage() {
+		this.languageJson = i18n.getLocale() || { '': {} };
+		const { localeSlug, localeVariant } = this.languageJson[ '' ];
+		this.localeCode = localeVariant || localeSlug;
+		this.currentLocale = find( languages, lang => lang.langSlug === this.localeCode );
+	}
+
+	/**
+	 * TODO: expand jsdoc comment
+	 *
+	 */
+	refresh = () => {
+		if ( this.initialized ) {
+			return;
+		}
+
+		if ( ! userSettings.getSettings() ) {
+			debug( 'initialization failed because userSettings are not ready' );
+			return;
+		}
+
+		if ( ! isCommunityTranslatorEnabled() ) {
+			debug( 'not initializing, not enabled' );
+			return;
+		}
+
+		this.setLanguage();
+
+		if ( ! this.localeCode || ! this.languageJson ) {
+			debug( 'trying to initialize translator without loaded language' );
+			return;
+		}
+
+		debug( 'Successfully initialized' );
+		this.initialized = true;
+	};
+
+	/**
+	 * Wraps translation in a DOM object and attaches `toString()` method in case in can't be rendered
+	 * @param { String } originalFromPage - original string
+	 * @param { String } displayedTranslationFromPage - translated string
+	 * @param  { Object } optionsFromPage - i18n.translate options
+	 * @returns {Object} DOM object
+	 */
+	wrapTranslation( originalFromPage, displayedTranslationFromPage, optionsFromPage ) {
+		if ( ! isCommunityTranslatorEnabled() ) {
+			return displayedTranslationFromPage;
+		}
+
+		if ( 'object' !== typeof optionsFromPage ) {
+			optionsFromPage = {};
+		}
+
+		if ( 'string' !== typeof originalFromPage ) {
+			debug( 'unknown original format' );
+			return displayedTranslationFromPage;
+		}
+
+		if ( 'boolean' === typeof optionsFromPage.textOnly && optionsFromPage.textOnly ) {
```

Comment:
Is there a case that ` !! optionsFromPage.textOnly` would be different from this?

Reply:
`!! optionsFromPage.textOnly` looks like it will work. From what I can see, when `textOnly` is set, it's only ever passed as a `Bool` with a value of `true`.